src/WrappedHttpHandler.php

```diff
@@ -138,11 +139,15 @@ private function parseError(
         if (!isset($err['response'])) {
             $parts = ['response' => null];
         } else {
-            $errorParser = $this->errorParser;
-            $parts = $errorParser($err['response']);
+            try {
+                $parts = call_user_func($this->errorParser, $err['response']);
+                $serviceError .= " {$parts['code']} ({$parts['type']}): "
+                    . "{$parts['message']} - " . $err['response']->getBody();
+            } catch (ParserException $e) {
+                $parts = [];
```

Comment:
Is there any way to add some kind of mention of the parse error in the returned error?